<a href="https://colab.research.google.com/github/Areeb-07/WiDS-DBA/blob/main/ML_models.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importing Libraries

In [ ]:
import pandas as pd
from sklearn import preprocessing
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.naive_bayes import GaussianNB
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import ExtraTreesClassifier
from sklearn.svm import LinearSVC
from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import AdaBoostClassifier
from sklearn.ensemble import GradientBoostingClassifier
import warnings
warnings.filterwarnings('ignore')

# Getting Data

In [ ]:
from google.colab import files
uploaded = files.upload() # The data will be loaded from the local device

Saving full_data_carla.csv to full_data_carla.csv


In [ ]:
df=pd.read_csv('full_data_carla.csv') # Reading the data into a DataFrame named df

In [ ]:
df # A look at the dataset

Unnamed: 0    accelX        accelY  ...     gyroX     gyroY     gyroZ
0               0 -0.244069  8.094744e-02  ...  0.000000  0.000000  0.000000
1               1 -0.206027 -9.387732e-07  ...  0.000000  0.000000  0.000000
2               2 -0.226992 -3.950313e-02  ...  0.000000  0.000000  0.000000
3               3 -0.225045  4.047326e-02  ...  0.000000  0.000000  0.000000
4               4 -0.226992 -3.950313e-02  ...  0.000000  0.000000  0.000000
...           ...       ...           ...  ...       ...       ...       ...
99353       99353 -1.511383  2.782583e-02  ...  0.008643 -0.001575 -0.005514
99354       99354 -1.494224  1.084785e-01  ...  0.007890 -0.001658 -0.005501
99355       99355 -1.523028 -2.127216e-01  ...  0.005966 -0.001740 -0.005473
99356       99356 -1.511467  2.783418e-02  ...  0.004825 -0.001693 -0.005456
99357       99357 -1.434334  1.105895e-01  ...  0.002422 -0.001670 -0.005421

[99358 rows x 8 columns]

-   The column 'Unnamed: 0' is unnecessary

In [ ]:
df.drop(['Unnamed: 0'],axis=1,inplace=True) # Dropping the 'Unnamed: 0' Column

In [ ]:
df.info() # Basic information about the dataset

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99358 entries, 0 to 99357
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   accelX  99358 non-null  float64
 1   accelY  99358 non-null  float64
 2   accelZ  99358 non-null  float64
 3   class   99358 non-null  object 
 4   gyroX   99358 non-null  float64
 5   gyroY   99358 non-null  float64
 6   gyroZ   99358 non-null  float64
dtypes: float64(6), object(1)
memory usage: 5.3+ MB


# Classifying the Driver Behaviour

### Creating Separate Dataframes for each Class

In [ ]:
apo=df[(df['class']=='apo')].reset_index(drop=True)
onder=df[(df['class']=='onder')].reset_index(drop=True)
hurcan=df[(df['class']=='hurcan')].reset_index(drop=True)
gonca=df[(df['class']=='gonca')].reset_index(drop=True)
mehdi=df[(df['class']=='mehdi')].reset_index(drop=True)
berk=df[(df['class']=='berk')].reset_index(drop=True)
selin=df[(df['class']=='selin')].reset_index(drop=True)

### Finding the First and Third Quartile for each Column of each Dataframe

 ##### Driver Apo

In [ ]:
apo_ax_1stquar=apo['accelX'].quantile(0.25)
apo_ax_3rdquar=apo['accelX'].quantile(0.75)
apo_ay_1stquar=apo['accelY'].quantile(0.25)
apo_ay_3rdquar=apo['accelY'].quantile(0.75)
apo_az_1stquar=apo['accelZ'].quantile(0.25)
apo_az_3rdquar=apo['accelZ'].quantile(0.75)
apo_gx_1stquar=apo['gyroX'].quantile(0.25)
apo_gx_3rdquar=apo['gyroX'].quantile(0.75)
apo_gy_1stquar=apo['gyroY'].quantile(0.25)
apo_gy_3rdquar=apo['gyroY'].quantile(0.75)
apo_gz_1stquar=apo['gyroZ'].quantile(0.25)
apo_gz_3rdquar=apo['gyroZ'].quantile(0.75)

#####Diver Onder

In [ ]:
onder_ax_1stquar=onder['accelX'].quantile(0.25)
onder_ax_3rdquar=onder['accelX'].quantile(0.75)
onder_ay_1stquar=onder['accelY'].quantile(0.25)
onder_ay_3rdquar=onder['accelY'].quantile(0.75)
onder_az_1stquar=onder['accelZ'].quantile(0.25)
onder_az_3rdquar=onder['accelZ'].quantile(0.75)
onder_gx_1stquar=onder['gyroX'].quantile(0.25)
onder_gx_3rdquar=onder['gyroX'].quantile(0.75)
onder_gy_1stquar=onder['gyroY'].quantile(0.25)
onder_gy_3rdquar=onder['gyroY'].quantile(0.75)
onder_gz_1stquar=onder['gyroZ'].quantile(0.25)
onder_gz_3rdquar=onder['gyroZ'].quantile(0.75)

#####Driver Hurcan

In [ ]:
hurcan_ax_1stquar=hurcan['accelX'].quantile(0.25)
hurcan_ax_3rdquar=hurcan['accelX'].quantile(0.75)
hurcan_ay_1stquar=hurcan['accelY'].quantile(0.25)
hurcan_ay_3rdquar=hurcan['accelY'].quantile(0.75)
hurcan_az_1stquar=hurcan['accelZ'].quantile(0.25)
hurcan_az_3rdquar=hurcan['accelZ'].quantile(0.75)
hurcan_gx_1stquar=hurcan['gyroX'].quantile(0.25)
hurcan_gx_3rdquar=hurcan['gyroX'].quantile(0.75)
hurcan_gy_1stquar=hurcan['gyroY'].quantile(0.25)
hurcan_gy_3rdquar=hurcan['gyroY'].quantile(0.75)
hurcan_gz_1stquar=hurcan['gyroZ'].quantile(0.25)
hurcan_gz_3rdquar=hurcan['gyroZ'].quantile(0.75)

#####Driver Gonca

In [ ]:
gonca_ax_1stquar=gonca['accelX'].quantile(0.25)
gonca_ax_3rdquar=gonca['accelX'].quantile(0.75)
gonca_ay_1stquar=gonca['accelY'].quantile(0.25)
gonca_ay_3rdquar=gonca['accelY'].quantile(0.75)
gonca_az_1stquar=gonca['accelZ'].quantile(0.25)
gonca_az_3rdquar=gonca['accelZ'].quantile(0.75)
gonca_gx_1stquar=gonca['gyroX'].quantile(0.25)
gonca_gx_3rdquar=gonca['gyroX'].quantile(0.75)
gonca_gy_1stquar=gonca['gyroY'].quantile(0.25)
gonca_gy_3rdquar=gonca['gyroY'].quantile(0.75)
gonca_gz_1stquar=gonca['gyroZ'].quantile(0.25)
gonca_gz_3rdquar=gonca['gyroZ'].quantile(0.75)

#####Driver Mehdi

In [ ]:
mehdi_ax_1stquar=mehdi['accelX'].quantile(0.25)
mehdi_ax_3rdquar=mehdi['accelX'].quantile(0.75)
mehdi_ay_1stquar=mehdi['accelY'].quantile(0.25)
mehdi_ay_3rdquar=mehdi['accelY'].quantile(0.75)
mehdi_az_1stquar=mehdi['accelZ'].quantile(0.25)
mehdi_az_3rdquar=mehdi['accelZ'].quantile(0.75)
mehdi_gx_1stquar=mehdi['gyroX'].quantile(0.25)
mehdi_gx_3rdquar=mehdi['gyroX'].quantile(0.75)
mehdi_gy_1stquar=mehdi['gyroY'].quantile(0.25)
mehdi_gy_3rdquar=mehdi['gyroY'].quantile(0.75)
mehdi_gz_1stquar=mehdi['gyroZ'].quantile(0.25)
mehdi_gz_3rdquar=mehdi['gyroZ'].quantile(0.75)

#####Driver Berk

In [ ]:
berk_ax_1stquar=berk['accelX'].quantile(0.25)
berk_ax_3rdquar=berk['accelX'].quantile(0.75)
berk_ay_1stquar=berk['accelY'].quantile(0.25)
berk_ay_3rdquar=berk['accelY'].quantile(0.75)
berk_az_1stquar=berk['accelZ'].quantile(0.25)
berk_az_3rdquar=berk['accelZ'].quantile(0.75)
berk_gx_1stquar=berk['gyroX'].quantile(0.25)
berk_gx_3rdquar=berk['gyroX'].quantile(0.75)
berk_gy_1stquar=berk['gyroY'].quantile(0.25)
berk_gy_3rdquar=berk['gyroY'].quantile(0.75)
berk_gz_1stquar=berk['gyroZ'].quantile(0.25)
berk_gz_3rdquar=berk['gyroZ'].quantile(0.75)

#####Driver Selin

In [ ]:
selin_ax_1stquar=selin['accelX'].quantile(0.25)
selin_ax_3rdquar=selin['accelX'].quantile(0.75)
selin_ay_1stquar=selin['accelY'].quantile(0.25)
selin_ay_3rdquar=selin['accelY'].quantile(0.75)
selin_az_1stquar=selin['accelZ'].quantile(0.25)
selin_az_3rdquar=selin['accelZ'].quantile(0.75)
selin_gx_1stquar=selin['gyroX'].quantile(0.25)
selin_gx_3rdquar=selin['gyroX'].quantile(0.75)
selin_gy_1stquar=selin['gyroY'].quantile(0.25)
selin_gy_3rdquar=selin['gyroY'].quantile(0.75)
selin_gz_1stquar=selin['gyroZ'].quantile(0.25)
selin_gz_3rdquar=selin['gyroZ'].quantile(0.75)

###Creating a new Column 'Behaviour' for each Dataframe

In [ ]:
apo['Behaviour']=apo['class']
onder['Behaviour']=onder['class']
hurcan['Behaviour']=hurcan['class']
gonca['Behaviour']=gonca['class']
mehdi['Behaviour']=mehdi['class']
berk['Behaviour']=berk['class']
selin['Behaviour']=selin['class']

#####This Column will be the target variable

####Assigning the Value 'Normal' to 'Behaviour' Column

In [ ]:
apo['Behaviour']='Normal'
onder['Behaviour']='Normal'
hurcan['Behaviour']='Normal'
gonca['Behaviour']='Normal'
mehdi['Behaviour']='Normal'
berk['Behaviour']='Normal'
selin['Behaviour']='Normal'

###Classifying columns as 'Drowsy'

In [ ]:
apo.loc[(apo['accelX']<apo_ax_1stquar) & (apo['accelY']<apo_ay_1stquar) & (apo['accelZ']<apo_az_1stquar) & (apo['gyroX']<apo_gx_1stquar) & (apo['gyroY']<apo_gy_1stquar) & (apo['gyroZ']<apo_gz_1stquar),'Behaviour']='Drowsy'
onder.loc[(onder['accelX']<onder_ax_1stquar) & (onder['accelY']<onder_ay_1stquar) & (onder['accelZ']<onder_az_1stquar) & (onder['gyroX']<onder_gx_1stquar) & (onder['gyroY']<onder_gy_1stquar) & (onder['gyroZ']<onder_gz_1stquar),'Behaviour']='Drowsy'
hurcan.loc[(hurcan['accelX']<hurcan_ax_1stquar) & (hurcan['accelY']<hurcan_ay_1stquar) & (hurcan['accelZ']<hurcan_az_1stquar) & (hurcan['gyroX']<hurcan_gx_1stquar) & (hurcan['gyroY']<hurcan_gy_1stquar) & (hurcan['gyroZ']<hurcan_gz_1stquar),'Behaviour']='Drowsy' 
gonca.loc[(gonca['accelX']<gonca_ax_1stquar) & (gonca['accelY']<gonca_ay_1stquar) & (gonca['accelZ']<gonca_az_1stquar) & (gonca['gyroX']<gonca_gx_1stquar) & (gonca['gyroY']<gonca_gy_1stquar) & (gonca['gyroZ']<gonca_gz_1stquar),'Behaviour']='Drowsy'
mehdi.loc[(mehdi['accelX']<mehdi_ax_1stquar) & (mehdi['accelY']<mehdi_ay_1stquar) & (mehdi['accelZ']<mehdi_az_1stquar) & (mehdi['gyroX']<mehdi_gx_1stquar) & (mehdi['gyroY']<mehdi_gy_1stquar) & (mehdi['gyroZ']<mehdi_gz_1stquar) ,'Behaviour']='Drowsy'
berk.loc[(berk['accelX']<berk_ax_1stquar) & (berk['accelY']<berk_ay_1stquar) & (berk['accelZ']<berk_az_1stquar) & (berk['gyroX']<berk_gx_1stquar) & (berk['gyroY']<berk_gy_1stquar) & (berk['gyroZ']<berk_gz_1stquar) ,'Behaviour']='Drowsy'
selin.loc[(selin['accelX']<selin_ax_1stquar) & (selin['accelY']<selin_ay_1stquar) & (selin['accelZ']<selin_az_1stquar) & (selin['gyroX']<selin_gx_1stquar) & (selin['gyroY']<selin_gy_1stquar) & (selin['gyroZ']<selin_gz_1stquar),'Behaviour']='Drowsy'

##### If the value of each column is less than the respective first quartile (for the respective class) for a record, then the behaviour 'Drowsy' is assigned to it

###Classifying Columns as 'Aggressive'

In [ ]:
apo.loc[(apo['accelX']>apo_ax_3rdquar) | (apo['accelY']>apo_ay_3rdquar) | (apo['accelZ']>apo_az_3rdquar) | (apo['gyroX']>apo_gx_3rdquar) | (apo['gyroY']>apo_gy_3rdquar) | (apo['gyroZ']>apo_gz_3rdquar),'Behaviour']='Aggressive'
onder.loc[(onder['accelX']>onder_ax_3rdquar) | (onder['accelY']>onder_ay_3rdquar) | (onder['accelZ']>onder_az_3rdquar) | (onder['gyroX']>onder_gx_3rdquar) | (onder['gyroY']>onder_gy_3rdquar) | (onder['gyroZ']>onder_gz_3rdquar),'Behaviour']='Aggressive'
hurcan.loc[(hurcan['accelX']>hurcan_ax_3rdquar) | (hurcan['accelY']>hurcan_ay_3rdquar) | (hurcan['accelZ']>hurcan_az_3rdquar) | (hurcan['gyroX']>hurcan_gx_3rdquar) | (hurcan['gyroY']>hurcan_gy_3rdquar) | (hurcan['gyroZ']>hurcan_gz_3rdquar),'Behaviour']='Aggressive'
gonca.loc[(gonca['accelX']>gonca_ax_3rdquar) | (gonca['accelY']>gonca_ay_3rdquar) | (gonca['accelZ']>gonca_az_3rdquar) | (gonca['gyroX']>gonca_gx_3rdquar) | (gonca['gyroY']>gonca_gy_3rdquar) | (gonca['gyroZ']>gonca_gz_3rdquar),'Behaviour']='Aggressive'
mehdi.loc[(mehdi['accelX']>mehdi_ax_3rdquar) | (mehdi['accelY']>mehdi_ay_3rdquar) | (mehdi['accelZ']>mehdi_az_3rdquar) | (mehdi['gyroX']>mehdi_gx_3rdquar) | (mehdi['gyroY']>mehdi_gy_3rdquar) | (mehdi['gyroZ']>mehdi_gz_3rdquar) ,'Behaviour']='Aggressive'
berk.loc[(berk['accelX']>berk_ax_3rdquar) | (berk['accelY']>berk_ay_3rdquar) | (berk['accelZ']>berk_az_3rdquar) | (berk['gyroX']>berk_gx_3rdquar) | (berk['gyroY']>berk_gy_3rdquar) | (berk['gyroZ']>berk_gz_3rdquar) ,'Behaviour']='Aggressive'
selin.loc[(selin['accelX']>selin_ax_3rdquar) | (selin['accelY']>selin_ay_3rdquar) | (selin['accelZ']>selin_az_3rdquar) | (selin['gyroX']>selin_gx_3rdquar) | (selin['gyroY']>selin_gy_3rdquar) | (selin['gyroZ']>selin_gz_3rdquar),'Behaviour']='Aggressive'

#####If the value of any column is more than the respective third quartile (for the respective class) for a record, then the behaviour 'Aggressive' is assigned to it

### Conactinating the Dataframes for each Class to a Single Dataframe Result

In [ ]:
frames=[apo,onder,hurcan,gonca,mehdi,berk,selin]
result=pd.concat(frames)

In [ ]:
result.reset_index(drop=True,inplace=True) # Resetting the index of the dataframe

In [ ]:
result # A glimpse at the dataframe

accelX        accelY    accelZ  ...     gyroY     gyroZ   Behaviour
0     -0.244069  8.094744e-02  9.807350  ...  0.000000  0.000000      Normal
1     -0.206027 -9.387732e-07  9.807837  ...  0.000000  0.000000      Normal
2     -0.226992 -3.950313e-02  9.807553  ...  0.000000  0.000000      Normal
3     -0.225045  4.047326e-02  9.807437  ...  0.000000  0.000000      Normal
4     -0.226992 -3.950313e-02  9.807553  ...  0.000000  0.000000      Normal
...         ...           ...       ...  ...       ...       ...         ...
99353 -1.511383  2.782583e-02  9.805461  ... -0.001575 -0.005514      Normal
99354 -1.494224  1.084785e-01  9.805621  ... -0.001658 -0.005501  Aggressive
99355 -1.523028 -2.127216e-01  9.804395  ... -0.001740 -0.005473      Normal
99356 -1.511467  2.783418e-02  9.805650  ... -0.001693 -0.005456      Normal
99357 -1.434334  1.105895e-01  9.805491  ... -0.001670 -0.005421  Aggressive

[99358 rows x 8 columns]

### Encoding the DataFrame

In [ ]:
a1=result['class'].unique()
a2=result['Behaviour'].unique()

#####The 'Object' type columns are made 'Numerical' type by Encoding

In [ ]:
encoder=preprocessing.LabelEncoder()
result['class']=encoder.fit_transform(result['class'])
result['Behaviour']=encoder.fit_transform(result['Behaviour'])

In [ ]:
result

accelX        accelY    accelZ  ...     gyroY     gyroZ  Behaviour
0     -0.244069  8.094744e-02  9.807350  ...  0.000000  0.000000          2
1     -0.206027 -9.387732e-07  9.807837  ...  0.000000  0.000000          2
2     -0.226992 -3.950313e-02  9.807553  ...  0.000000  0.000000          2
3     -0.225045  4.047326e-02  9.807437  ...  0.000000  0.000000          2
4     -0.226992 -3.950313e-02  9.807553  ...  0.000000  0.000000          2
...         ...           ...       ...  ...       ...       ...        ...
99353 -1.511383  2.782583e-02  9.805461  ... -0.001575 -0.005514          2
99354 -1.494224  1.084785e-01  9.805621  ... -0.001658 -0.005501          0
99355 -1.523028 -2.127216e-01  9.804395  ... -0.001740 -0.005473          2
99356 -1.511467  2.783418e-02  9.805650  ... -0.001693 -0.005456          2
99357 -1.434334  1.105895e-01  9.805491  ... -0.001670 -0.005421          0

[99358 rows x 8 columns]

#####The Encoded Features

In [ ]:
b1=result['class'].unique()
b2=result['Behaviour'].unique()
classes={}
Behaviour={}
for i in range(len(a1)):
    classes[a1[i]]=b1[i]
for i in range(len(a2)):
    Behaviour[a2[i]]=b2[i]
print(classes)
print(Behaviour)

{'apo': 0, 'onder': 5, 'hurcan': 3, 'gonca': 2, 'mehdi': 4, 'berk': 1, 'selin': 6}
{'Normal': 2, 'Aggressive': 0, 'Drowsy': 1}


#####The original values of the class and Behaviour columns aginst their encoded values

# Applying Machine Learning Models

###Performing train_test_split

In [ ]:
data=result.values # Converts DataFrame to array which can be taken as argument in train_test_split

In [ ]:
# split into inputs and outputs
X, y = data[:, :-1], data[:, -1]
print(X.shape, y.shape)
# split into train test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=1)
print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)

(99358, 7) (99358,)
(66569, 7) (32789, 7) (66569,) (32789,)


- X_train and y_train will be used to train the models
- X_test will be used to test the models 
- y_test will be used to calculate the accuracy of the models

## Random Forest Classifier

In [ ]:
# fit the model
model = RandomForestClassifier(random_state=1)
model.fit(X_train, y_train)

RandomForestClassifier(random_state=1)

In [ ]:
# make predictions
yhat = model.predict(X_test)

In [ ]:
# evaluate predictions
acc = accuracy_score(y_test, yhat)
accuracy=acc*100
print('Accuracy: %.3f' % accuracy)

Accuracy: 99.396


## Gaussian Naive Bayes Model

In [ ]:
# training the model on training set
gnb = GaussianNB()
gnb.fit(X_train, y_train)

GaussianNB()

In [ ]:
# making predictions on the testing set
y_pred = gnb.predict(X_test)

In [ ]:
# comparing actual response values (y_test) with predicted response values (y_pred)
print("Gaussian Naive Bayes model accuracy(in %):",accuracy_score(y_test, y_pred)*100)

Gaussian Naive Bayes model accuracy(in %): 36.50614535362469


## Logistic Regresssion Model

In [ ]:
# instantiate the model (using the default parameters)
logreg = LogisticRegression()
# fit the model with data
logreg.fit(X_train,y_train)

LogisticRegression()

In [ ]:
# Making predictions on the testing set
y_pred=logreg.predict(X_test)

In [ ]:
# comparing actual response values (y_test) with predicted response values (y_pred)
print("Logistic Regression model accuracy(in %):",accuracy_score(y_test, y_pred)*100)

Logistic Regression model accuracy(in %): 77.69678855713806


## Extra Trees Classifier

In [ ]:
# Fit the model
extree=ExtraTreesClassifier()
extree.fit(X_train,y_train)

ExtraTreesClassifier()

In [ ]:
# Make Prediction
y_pred=extree.predict(X_test)

In [ ]:
# evaluate predictions
acc = accuracy_score(y_test, y_pred)
accuracy=acc*100
print('Accuracy: %.3f' % accuracy)

Accuracy: 97.310


### Scaling the Training and Testing Sets

In [ ]:
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

## Linear Support Vector Classifier

In [ ]:
# Training the model on training set
lsvc = LinearSVC(verbose=0)
lsvc.fit(X_train, y_train)
# Making prediction on the testing set
ypred = lsvc.predict(X_test)

In [ ]:
# comparing actual response values (y_test) with predicted response values (y_pred)
print("Linear SVC model accuracy(in %):",accuracy_score(y_test, ypred)*100)

Linear SVC model accuracy(in %): 78.0170179023453


## AdaBoost Classifier

### AdaBoost with Decision Tree Base Estimator

In [ ]:
# Create adaboost classifer object
abc = AdaBoostClassifier(n_estimators=50,learning_rate=1) # The default Base Estimator is Decision Tree
# Train Adaboost Classifer
model = abc.fit(X_train, y_train)
#Predict the response for test dataset
y_pred = model.predict(X_test)

In [ ]:
# comparing actual response values (y_test) with predicted response values (y_pred)
print("Accuracy(in %):",accuracy_score(y_test, y_pred)*100)

Accuracy(in %): 75.98584891274513


## Stochastic Gradient Boosting Classification

In [ ]:
# Fit the Model
gbc = GradientBoostingClassifier(n_estimators=50)
# Train Adaboost Classifer
model = gbc.fit(X_train, y_train)
#Predict the response for test dataset
y_pred = model.predict(X_test)

In [ ]:
# comparing actual response values (y_test) with predicted response values (y_pred)
print("Accuracy(in %):",accuracy_score(y_test, y_pred)*100)

Accuracy(in %): 97.42596602519139


# Conclusion

### The Accuracies of the Various Models

- Random Forest Classifier: 99.396%
- Gaussian Naive Bayes Model: 36.506%
- Logistic Regression Model: 77.697%
- Extra Trees Classifier: 97.310%
- Linear Support Vector Classifier: 78.017%
- AdaBoost Classifier with Decision Tree: 75.986%
- Stochastic Gradient Boosting Classifier: 97.426%